atom/browser/browser.cc

```diff
@@ -152,6 +164,33 @@ bool Browser::HandleBeforeQuit() {
   return !prevent_default;
 }
 
+void Browser::InitializeSingleInstance() {
+  base::FilePath userDir;
+  PathService::Get(brightray::DIR_USER_DATA, &userDir);
+
+  auto no_refcount_this = base::Unretained(this);
+  process_singleton_.reset(new AtomProcessSingleton(
+    userDir,
+    base::Bind(&Browser::OnProcessSingletonNotification, no_refcount_this)));
```

Comment:
We can pass the callback in `InitializeSingleInstance` instead of adding a `SetSingleInstanceCallback` method, and `OnProcessSingletonNotification` is also not necessary since we can pass the callback directly to `AtomProcessSingleton`.

Also I think we don't need to touch `Browser` too, all these can be moved to `App` class.